### PR TITLE
Clean up Operate dependencies

### PR DESCRIPTION
**Pull request by ChrisKujawa**
## Description

This is the first step into the Single Jar/Application. In order to be able to make progress we started to apply the maven dependency and enforcer plugin. This can be seen as a preparation before inheriting from the Zeebe parent.

These plugins allow us to detect duplicate dependencies, dependencies which are not used, dependencies that are used but not declared (transient dependencies).

Based on the plugins we cleaned up all Operate pom's. We have defined as part of this PR
all necessary dependencies and their versions in the Operate root/parent pom. This gives
a better overview and also makes it easier later to move the dependency declarations up to the Zeebe parent (if necessary). 

## Related issues

related to #17009

### CODE REVIEW

File: operate/qa/data-generator/pom.xml [hunk 1]
```diff
@@ -36,24 +34,50 @@
...
-
-    <plugins>
-
-      <!-- LICENSE PLUGIN -->
-      <plugin>
-        <groupId>com.mycila</groupId>
-        <artifactId>license-maven-plugin</artifactId>
-      </plugin>
-    </plugins>
-  </build>
```
Comment (Zelldon) [inline operate/qa/data-generator/pom.xml:57]: ❌ This needs to be reverted

File: operate/webapp/pom.xml [hunk 2]
```diff
@@ -8,55 +8,64 @@
...
 
   <name>Operate Webapp</name>
-  <artifactId>operate-webapp</artifactId>
 
   <dependencies>
-
     <dependency>
       <groupId>io.camunda</groupId>
       <artifactId>operate-common</artifactId>
       <version>${project.version}</version>
```
Comment (houssain-barouni) [inline operate/webapp/pom.xml:19]: same we can remove version here

File: operate/webapp/pom.xml [hunk 3]
```diff
@@ -8,55 +8,64 @@
...
     </dependency>
-
     <dependency>
       <groupId>io.camunda</groupId>
       <artifactId>operate-schema</artifactId>
       <version>${project.version}</version>
```
Comment (houssain-barouni) [inline operate/webapp/pom.xml:24]: same we can remove version here

File: operate/webapp/pom.xml [hunk 4]
```diff
@@ -8,55 +8,64 @@
...
     </dependency>
-
     <dependency>
       <groupId>io.camunda</groupId>
       <artifactId>operate-data-generator</artifactId>
       <version>${project.version}</version>
```
Comment (houssain-barouni) [inline operate/webapp/pom.xml:29]: same we can remove version here

File: operate/webapp/pom.xml [hunk 5]
```diff
@@ -8,55 +8,64 @@
...
+      <scope>runtime</scope>
     </dependency>
-
     <dependency>
       <groupId>io.camunda</groupId>
       <artifactId>operate-importer</artifactId>
       <version>${project.version}</version>
```
Comment (houssain-barouni) [inline operate/webapp/pom.xml:35]: same we can remove version here

File: operate/webapp/pom.xml [hunk 6]
```diff
@@ -8,55 +8,64 @@
...
+      <scope>runtime</scope>
     </dependency>
-
     <dependency>
       <groupId>io.camunda</groupId>
       <artifactId>operate-archiver</artifactId>
       <version>${project.version}</version>
```
Comment (houssain-barouni) [inline operate/webapp/pom.xml:41]: same we can remove version here

File: operate/webapp/pom.xml [hunk 7]
```diff
@@ -8,55 +8,64 @@
...
+      <scope>runtime</scope>
     </dependency>
-
     <dependency>
-      <groupId>org.junit.jupiter</groupId>
-      <artifactId>junit-jupiter-engine</artifactId>
-      <scope>test</scope>
+      <groupId>io.camunda</groupId>
+      <artifactId>operate-webjar</artifactId>
+      <version>${project.version}</version>
```
Comment (houssain-barouni) [inline operate/webapp/pom.xml:47]: same we can remove version here

File: operate/webapp/pom.xml [hunk 8]
```diff
@@ -8,55 +8,64 @@
...
+    </dependency>
+    <!-- OAuth0 -->
+    <dependency>
+      <groupId>io.camunda</groupId>
+      <artifactId>operate-mvc-auth-commons</artifactId>
+      <version>${project.version}</version>
```
Comment (houssain-barouni) [inline operate/webapp/pom.xml:53]: same we can remove version here

**Review by houssain-barouni — COMMENTED** [review]
Just few comments about unnecessary dependency version declaration.
I wonder if we should park this until after the minor release 🤔 ?

**Comment by Zelldon** [comment]
Fun fact: I removed all the versions before 😅🙈 bit after copy pasting in the morning again some of the change seems to be back again. I will take care of it. :+1:

**Comment by houssain-barouni** [comment]
@Zelldon E2E tests are failing with this error 🤔 
```
Failed to execute goal org.eclipse.transformer:transformer-maven-plugin:0.5.0:jar (source-jar) on project operate-mvc-auth-commons: Unable to resolve artifact com.auth0:mvc-auth-commons:sources:1.11.0:jar excludes=nul
```

**Comment by Zelldon** [comment]
@houssain-barouni I think this was a fluke, after retry it worked.

**Review by romansmirnov — COMMENTED** [review]
When building a distro and starting it, it "complains" that there are multiple SL4J providers on the classpath:

```
SLF4J(W): Class path contains multiple SLF4J providers.
SLF4J(W): Found provider [ch.qos.logback.classic.spi.LogbackServiceProvider@17579e0f]
SLF4J(W): Found provider [org.apache.logging.slf4j.SLF4JServiceProvider@4d41cee]
SLF4J(W): See https://www.slf4j.org/codes.html#multiple_bindings for an explanation.
SLF4J(I): Actual provider is of type [ch.qos.logback.classic.spi.LogbackServiceProvider@17579e0f]
ERROR StatusLogger Cannot set JUL log level through log4j-api: ignoring call to Logger.setLevel(SEVERE)
ERROR StatusLogger Cannot set JUL log level through log4j-api: ignoring call to Logger.setLevel(SEVERE)
ERROR StatusLogger Cannot set JUL log level through log4j-api: ignoring call to Logger.setLevel(WARNING)
ERROR StatusLogger Cannot set JUL log level through log4j-api: ignoring call to Logger.setLevel(WARNING)
ERROR StatusLogger Cannot set JUL log level through log4j-api: ignoring call to Logger.setLevel(WARNING)
ERROR StatusLogger Cannot set JUL log level through log4j-api: ignoring call to Logger.setLevel(SEVERE)
ERROR StatusLogger Cannot set JUL log level through log4j-api: ignoring call to Logger.setLevel(WARNING)
ERROR StatusLogger Cannot set JUL log level through log4j-api: ignoring call to Logger.setLevel(WARNING)
ERROR StatusLogger Cannot set JUL log level through log4j-api: ignoring call to Logger.setLevel(INFO)
```

Compared to the previous distribution, this was not the case.

When comparing a previous distro with this "new" one, I noticed the following:
* `aws-java-sdk-sts` is missing on the classpath in the "new" distribution. I guess it is not used that's why it is removed as a dependency, but let's double check if that's okay.
* `tomcat-annotations-api` is added to the classpath in the "new" distribution. The problem that it collides with `jakarta.annotation-api` having the same classes/annotations. I would suggest excluding `tomcat-annotations-api` as a dependency.

Next step: I am going through the changes in the pom.xmls

**Comment by Zelldon** [comment]
@romansmirnov I fixed the issues you brought up. But it looks like we miss somehow the skyscreamer dependency, I have no clue where this is coming from. 

https://github.com/camunda/zeebe/actions/runs/8392843163/job/22986430211?pr=17031
```
Error:  io.camunda.operate.rest.HealthCheckIT.testReadyStateIsNotOK -- Time elapsed: 0.003 s <<< ERROR!
java.lang.NoClassDefFoundError: org/skyscreamer/jsonassert/JSONAssert
	at org.springframework.test.util.JsonExpectationsHelper.assertJsonEqual(JsonExpectationsHelper.java:61)
	at org.springframework.test.web.servlet.result.ContentResultMatchers.lambda$json$9(ContentResultMatchers.java:227)
	at org.springframework.test.web.servlet.MockMvc$1.andExpect(MockMvc.java:214)
	at io.camunda.operate.rest.HealthCheckIT.testReadyStateIsNotOK(HealthCheckIT.java:95)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.springframework.test.context.junit4.statements.RunBeforeTestExecutionCallbacks.evaluate(RunBeforeTestExecutionCallbacks.java:76)
	at org.springframework.test.context.junit4.statements.RunAfterTestExecutionCallbacks.evaluate(RunAfterTestExecutionCallbacks.java:84)
	at org.springframework.test.context.junit4.statements.RunBeforeTestMethodCallbacks.evaluate(RunBeforeTestMethodCallbacks.java:75)
	at org.springframework.test.context.junit4.statements.RunAfterTestMethodCallbacks.evaluate(RunAfterTestMethodCallbacks.java:86)
	at org.springframework.test.context.junit4.statements.SpringRepeat.evaluate(SpringRepeat.java:84)
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.runChild(SpringJUnit4ClassRunner.java:252)
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.runChild(SpringJUnit4ClassRunner.java:97)
	at org.springframework.test.context.junit4.statements.RunBeforeTestClassCallbacks.evaluate(RunBeforeTestClassCallbacks.java:61)
	at org.springframework.test.context.junit4.statements.RunAfterTestClassCallbacks.evaluate(RunAfterTestClassCallbacks.java:70)
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.run(SpringJUnit4ClassRunner.java:191)
Caused by: java.lang.ClassNotFoundException: org.skyscreamer.jsonassert.JSONAssert
	... 15 more

**Comment by romansmirnov** [comment]
@Zelldon, let me check quickly.

**Comment by romansmirnov** [comment]
@Zelldon, the dependency comes with `spring-boot-starter-test`. I would suggest adding `spring-boot-starter-test` to `operate/qa/integration-tests/pom.xml`.

**Comment by Zelldon** [comment]
Thanks @romansmirnov 🙇🏼 Hm Idk why this was removed then? 🤔  I added it, let's see.

**Comment by romansmirnov** [comment]
@Zelldon, could it be that the LDAP test fails because of the missing LDAP test server dependency?

```
<dependency>
  <groupId>com.unboundid</groupId>
  <artifactId>unboundid-ldapsdk</artifactId>
  <version>6.0.11</version>
</dependency>
```

**Comment by Zelldon** [comment]
Sorry for that I guess something went wrong with merging the different PRs and poms, etc. I guess I have to verify again whether the dependency plugin is correctly configured and the dependencies are set... I will take a look at this. Sorry for the noise.

**Comment by romansmirnov** [comment]
I think you can revert https://github.com/camunda/zeebe/pull/17031/commits/2049b5d77e2afd0db90fcb29b81e7668748de911

**Review by romansmirnov — APPROVED** [review]
LGTM 🚀

**Pull request merged**